src/Encamina.Enmarcha.SemanticKernel/Extensions/KernelExtensions.cs

```diff
@@ -101,17 +100,51 @@ public static async Task<string> GetKernelFunctionPromptAsync(this Kernel kernel
             throw new InvalidOperationException(ExceptionMessages.ResourceManager.GetFormattedStringByCurrentCulture(nameof(ExceptionMessages.PromptEmbeddedResourcesNotFound), pluginName, function.Name, assembly.GetName().Name));
         }
 
-        return await InnerGetKernelFunctionPromptAsync(
+        return await GetKernelPromptAsync(
             kernel,
-            function.Name,
             ReadResource(assembly, promptTemplateResourceName),
-            ReadResource(assembly, promptConfigurationResourceName),
             arguments,
             promptTemplateFormat,
             promptTemplateFactory,
             cancellationToken);
     }
 
+    /// <summary>
+    /// Generates the final prompt for a given prompt using the arguments.
+    /// </summary>
+    /// <param name="kernel">The <see cref="Kernel"/> to work with.</param>
+    /// <param name="promptTemplate">The name of the plugin associated with the prompt function.</param>
+    /// <param name="arguments">The arguments passed to the function.</param>
+    /// <param name="promptTemplateFormat">
+    /// The template format for the prompt. This must be provided if <paramref name="promptTemplateFactory"/> is not <see langword="null"/>. This parameter is optional.
+    /// </param>
+    /// <param name="promptTemplateFactory">
+    /// A <see cref="IPromptTemplateFactory"/> to interpret the prompt and its configuration into a <see cref="IPromptTemplate"/>. This parameter is optional.
+    /// </param>
+    /// <param name="cancellationToken">A cancellation token that can be used to receive notice of cancellation.</param>
+    /// <returns>A string containing the generated prompt.</returns>
+    public static Task<string> GetKernelPromptAsync(this Kernel kernel, string promptTemplate, KernelArguments arguments, string promptTemplateFormat = null, IPromptTemplateFactory promptTemplateFactory = null, CancellationToken cancellationToken = default)
```

Comment:
You have to take into consideration the warnings generated on the new code by the integration process. Please check the following warning: Split this 257 characters long line (which is greater than 200 authorised). (https://rules.sonarsource.com/csharp/RSPEC-103)

Reply:
So, our ruleset in SonarCloud is a bit outdated. This warnings related to 200 characters per line is from times before wide screens, which almost all of us have. This one is safe to be ignored.